### PR TITLE
platform, common, lowlevel modules cleanup

### PR DESCRIPTION
**Pull request by cepetr**
This PR focuses on cleaning up tangled code in the `platform.c/h`, `common.c/h`, `lowlevel.c/h` modules, as well as the Unix emulator.

1. Removed some inappropriate dependencies (such as CPU clock setting and the `main_clean_exit` cyclic dependency) from the emulator code.
2. Most code from the `lowlevel` module has been relocated to the new `option_bytes` module.
3. Introduced a small `pvd` (Power Voltage Detector) module, which was previously split across two different places.
4. Introduced a small `reset_flags` module, which was previously located in the `lowlevel` module. Although it's currently not very useful, it might be improved in the future.
5. Most code from `platform.c` has been moved to a new `startup_init` module, which better reflects its purpose.
6. The `platform.h` header no longer exists, freeing up the name for potential future use.

Possible next steps:
1. `common.h` now only contains just utility functions.  We might consider renaming this module to better reflect its purpose.
2. The routines in `utils.s` seem slightly out of place and could be better organized between `bootutil` and `startup_init`.

### CODE REVIEW

File: core/embed/trezorhal/stm32f4/pvd.c [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  __HAL_RCC_PWR_CLK_ENABLE();
+  PWR_PVDTypeDef pvd_config = {0};
+  pvd_config.PVDLevel = PWR_PVDLEVEL_5;
+  pvd_config.Mode = PWR_PVD_MODE_IT_RISING_FALLING;
+  HAL_PWR_ConfigPVD(&pvd_config);
+  HAL_PWR_EnablePVD();
+#ifdef STM32U5
+  NVIC_EnableIRQ(PVD_PVM_IRQn);
+#else
+  NVIC_EnableIRQ(PVD_IRQn);
```
Comment (TychoVrahe) [inline core/embed/trezorhal/stm32f4/pvd.c:42]: maybe we should be explicit about IRQ priority

File: core/embed/trezorhal/stm32f4/pvd.c [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+#endif
+}
+
+void PVD_PVM_IRQHandler(void) {
```
Comment (TychoVrahe) [inline core/embed/trezorhal/stm32f4/pvd.c:46]: the correct name for F4 is PVD_IRQHandler, this issue originates from core-split but let's fix it here.

**Review by TychoVrahe — CHANGES_REQUESTED** [review]
Added few comments. 

There are also some problem with emulator builds due to additional usage of `#include STM32_HAL_H` that is not available for unix,  previously it was hidden in the platform.h.

File: core/embed/trezorhal/stm32f4/pvd.c (hunk 1 above)
Comment (cepetr) [reply core/embed/trezorhal/stm32f4/pvd.c:45]: I agree, fixed cbce0597696366abc7d32c967a3ae745b4d052ac

File: core/embed/models/T2B1/boards/trezor_r_v10.h [hunk 3]
```diff
@@ -7,6 +7,7 @@
 #define USE_SBU 1
 #define USE_I2C 1
 #define USE_CONSUMPTION_MASK 1
+#define USE_PVD
```
[the remark this answers is not in the document]
Comment (cepetr) [reply core/embed/models/T2B1/boards/trezor_r_v10.h]: fixed cbce0597696366abc7d32c967a3ae745b4d052ac

File: core/embed/trezorhal/stm32f4/pvd.c (hunk 2 above)
Comment (cepetr) [reply core/embed/trezorhal/stm32f4/pvd.c:50]: Yes, you are right, fixed cbce0597696366abc7d32c967a3ae745b4d052ac

**Comment by cepetr** [comment]
> Added few comments.
> 
> There are also some problem with emulator builds due to additional usage of `#include STM32_HAL_H` that is not available for unix, previously it was hidden in the platform.h.

fixed in 09652a435d301c1b39c3d32ba702ab82f3d4b659.

**Review by TychoVrahe — APPROVED** [review]

**Review by TychoVrahe — APPROVED** [review]

**Comment by mmilata** [comment]
Looks like this breaks `PRODUCTION=1` build (which we only do nightly) due to unknown reset flags:
```
embed/trezorhal/stm32f4/reset_flags.c: In function 'reset_flags_check':
embed/trezorhal/stm32f4/reset_flags.c:33:20: error: 'RCC_CSR_OBLRSTF' undeclared (first use in this function); did you mean 'RCC_CSR_BORRSTF'?
   33 |                    RCC_CSR_OBLRSTF)) != (RCC_CSR_PINRSTF | RCC_CSR_BORRSTF)) {
```
```
 embed/trezorhal/stm32u5/reset_flags.c: In function 'reset_flags_check':
embed/trezorhal/stm32u5/reset_flags.c:32:38: error: 'RCC_CSR_PORRSTF' undeclared (first use in this function); did you mean 'RCC_CSR_BORRSTF'?
   32 |                    RCC_CSR_SFTRSTF | RCC_CSR_PORRSTF | RCC_CSR_PINRSTF |
```
https://github.com/trezor/trezor-firmware/actions/runs/11467233095/job/31909674128

**Comment by cepetr** [comment]
> Looks like this breaks `PRODUCTION=1` build (which we only do nightly) due to unknown reset flags:
> 
> ```
> embed/trezorhal/stm32f4/reset_flags.c: In function 'reset_flags_check':
> embed/trezorhal/stm32f4/reset_flags.c:33:20: error: 'RCC_CSR_OBLRSTF' undeclared (first use in this function); did you mean 'RCC_CSR_BORRSTF'?
>    33 |                    RCC_CSR_OBLRSTF)) != (RCC_CSR_PINRSTF | RCC_CSR_BORRSTF)) {
> ```
> 
> ```
>  embed/trezorhal/stm32u5/reset_flags.c: In function 'reset_flags_check':
> embed/trezorhal/stm32u5/reset_flags.c:32:38: error: 'RCC_CSR_PORRSTF' undeclared (first use in this function); did you mean 'RCC_CSR_BORRSTF'?
>    32 |                    RCC_CSR_SFTRSTF | RCC_CSR_PORRSTF | RCC_CSR_PINRSTF |
> ```
> 
> https://github.com/trezor/trezor-firmware/actions/runs/11467233095/job/31909674128

Yes, unfortunately, I swapped the flags for U5 and F4. I fixed it in #4278. @mmilata

**Pull request merged**